node/core/backing/src/lib.rs

```diff
@@ -657,329 +656,97 @@ impl CandidateBackingJob {
 	}
 }
 
-struct JobHandle {
-	abort_handle: future::AbortHandle,
-	to_job: mpsc::Sender<ToJob>,
-	finished: oneshot::Receiver<()>,
-	su_handle: usize,
-}
-
-impl JobHandle {
-	async fn stop(mut self) {
-		let _ = self.to_job.send(ToJob::Stop).await;
-		let stop_timer = Delay::new(Duration::from_secs(1));
-
-		match future::select(stop_timer, self.finished).await {
-			Either::Left((_, _)) => {
-			},
-			Either::Right((_, _)) => {
-				self.abort_handle.abort();
-			},
-		}
-	}
-
-	async fn send_msg(&mut self, msg: ToJob) -> Result<(), Error> {
-		Ok(self.to_job.send(msg).await?)
-	}
-}
-
-struct Jobs<S> {
-	spawner: S,
-	running: HashMap<Hash, JobHandle>,
-	outgoing_msgs: StreamUnordered<mpsc::Receiver<FromJob>>,
-}
-
-async fn run_job(
-	parent: Hash,
-	keystore: KeyStorePtr,
-	rx_to: mpsc::Receiver<ToJob>,
-	mut tx_from: mpsc::Sender<FromJob>,
-) -> Result<(), Error> {
-	let (validators, roster) = futures::try_join!(
-		request_validators(parent, &mut tx_from).await?,
-		request_validator_groups(parent, &mut tx_from).await?,
-	)?;
-
-	let key = signing_key(&validators[..], &keystore).ok_or(Error::NotInValidatorSet)?;
-	let mut groups = HashMap::new();
-
-	for assignment in roster.scheduled {
-		if let Some(g) = roster.validator_groups.get(assignment.group_idx.0 as usize) {
-			groups.insert(
-				assignment.para_id,
-				g.clone(),
-			);
-		}
-	}
+impl util::JobTrait for CandidateBackingJob {
+	type ToJob = ToJob;
+	type FromJob = FromJob;
+	type Error = Error;
+	type RunArgs = KeyStorePtr;
 
-	let mut assignment = Default::default();
+	const NAME: &'static str = "CandidateBackingJob";
```

Comment:
looking from github view, this seems to be unused, could be a leftover after a refactoring

Reply:
https://github.com/paritytech/polkadot/blob/38ee247f96bbf3b7a70f2eebed827b80d04e7092/node/subsystem/src/util.rs#L355-L362

Not a huge deal, but it's used.

Reply:
ah, sorry then